source/lac/block_sparsity_pattern.cc

```diff
@@ -149,10 +145,10 @@ BlockSparsityPatternBase<SparsityPatternType>::collect_sizes()
 
 
   // then do the same with the columns
-  for (size_type c = 0; c < columns; ++c)
+  for (size_type c = 0; c < block_columns; ++c)
```

Comment:
For consistent code, should this be
```suggestion
  for (size_type c = 0; c < n_block_cols(); ++c)
```